### PR TITLE
Feature/apps 1722

### PR DESCRIPTION
**Pull request by HylandAditya**
MoveToRuleOnFoldersTest is converted .

### CODE REVIEW

File: amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/rules/FileAsRecordRuleTests.java [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+        RecordCategory recordCategory = new RecordCategory().builder()
+            .id(category_manager.getId()).build();
+        RecordCategoryChild folderWithRule = createFolder(recordCategory.getId(), getRandomName("recordFolder"));
+        RuleDefinition ruleDefinition = RuleDefinition.createNewRule().title("name").description("description")
+            .applyToChildren(true)
+            .actions(Collections.singletonList(ActionsOnRule.DECLARE_AS_RECORD.getActionValue()));
+        rulesAPI.createRule(getAdminUser().getUsername(), getAdminUser().getPassword(), NODE_PREFIX + folderWithRule.getId(), ruleDefinition);
+
+        STEP("Create as nonRMuser a new file into the previous folder in order to trigger the rule");
+        inPlaceRecord = dataContent.usingUser(nonRMUser).usingResource(testFolder).createContent(CMISUtil.DocumentType.TEXT_PLAIN);
```
Comment (kavitshah-gl) [inline amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/rules/FileAsRecordRuleTests.java:224]: There is no checks done to make sure the file is present in unfiled records. Need to complete the test case.

File: amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/rules/FileVersionAsRecordRuleTest.java [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+
+
+        STEP("Create a collaboration folder with a rule set to declare and file version as record to a record folder");
+        RecordCategoryChild folderWithRule = createFolder(recordCategory.getId(), getRandomName("recordFolder"));
+        RuleDefinition ruleDefinition = RuleDefinition.createNewRule().title("name").description("description")
+            .applyToChildren(true)
+            .actions(Collections.singletonList(ActionsOnRule.DECLARE_AS_RECORD.getActionValue()));
+        rulesAPI.createRule(getAdminUser().getUsername(), getAdminUser().getPassword(), NODE_PREFIX + folderWithRule.getId(), ruleDefinition);
+
```
Comment (kavitshah-gl) [inline amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/rules/FileVersionAsRecordRuleTest.java:117]: Verify if the rule has been created.

File: amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/rules/FileVersionAsRecordRuleTest.java [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+        STEP("Create a rule with Declare as Record action and check that user can select a record folder.");
+        RecordCategory recordCategory = new RecordCategory().builder()
+            .id(category_manager.getId()).build();
+        RecordCategoryChild folderWithRule = createFolder(recordCategory.getId(), getRandomName("recordFolder"));
+        RuleDefinition ruleDefinition = RuleDefinition.createNewRule().title("name").description("description")
+            .applyToChildren(true)
+            .actions(Collections.singletonList(ActionsOnRule.DECLARE_AS_RECORD.getActionValue()));
+        rulesAPI.createRule(getAdminUser().getUsername(), getAdminUser().getPassword(), NODE_PREFIX + folderWithRule.getId(), ruleDefinition);
+
```
Comment (kavitshah-gl) [inline amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/rules/FileVersionAsRecordRuleTest.java:137]: Record should be filed as present in enterprise-share test case.

File: amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/rules/FileVersionAsRecordRuleTest.java [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
```
Comment (kavitshah-gl) [inline amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/rules/FileVersionAsRecordRuleTest.java:159]: Rule should be created using Non Rm credentials.

Comment (kavitshah-gl) [reply amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/rules/FileVersionAsRecordRuleTest.java:159]: Record should be filed as mentioned in Share AGS test case

File: amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/rules/FileVersionAsRecordRuleTest.java [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+    public void triggerFileVersionToRecordFolderRuleAsUserWithPermissions()
+    {
+
+        STEP("Create as rmManager a new file into the folderWithRule in order to trigger the rule");
+        FileModel testFile = dataContent.usingUser(rmManager).usingResource(testFolder).createContent(CMISUtil.DocumentType.TEXT_PLAIN);
+
+    }
+
+    @Test
+    public void triggerFileVersionToRecordFolderRuleAsUserWithoutPermissions()
```
Comment (kavitshah-gl) [inline amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/rules/FileVersionAsRecordRuleTest.java:173]: Remove the test case if it is only UI specific test.

File: amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/rules/FileVersionAsRecordRuleTest.java [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+        RuleDefinition ruleDefinition = RuleDefinition.createNewRule().title("name").description("description")
+            .applyToChildren(true)
+            .actions(Collections.singletonList(ActionsOnRule.DECLARE_AS_RECORD.getActionValue()));
+        rulesAPI.createRule(getAdminUser().getUsername(), getAdminUser().getPassword(), NODE_PREFIX + folderWithRule.getId(), ruleDefinition);
+
+
+    }
+
+    @Test
+    public void triggerFileVersionToRecordFolderRuleAsUserWithPermissions()
```
Comment (kavitshah-gl) [inline amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/rules/FileVersionAsRecordRuleTest.java:164]: Remove the test case if it is only UI specific test.

File: amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/rules/FileVersionAsRecordRuleTest.java [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+        RecordCategory recordCategory = new RecordCategory().builder()
+            .id(category_manager.getId()).build();
+        RecordCategoryChild folderWithRule = createFolder(recordCategory.getId(), getRandomName("recordFolder"));
+        RuleDefinition ruleDefinition = RuleDefinition.createNewRule().title("name").description("description")
+            .applyToChildren(true)
+            .actions(Collections.singletonList(ActionsOnRule.DECLARE_AS_RECORD.getActionValue()));
+        rulesAPI.createRule(getAdminUser().getUsername(), getAdminUser().getPassword(), NODE_PREFIX + folderWithRule.getId(), ruleDefinition);
+
+        STEP("Create as nonRMuser a new file into the previous folder in order to trigger the rule");
+        inPlaceRecord = dataContent.usingUser(nonRMuser).usingResource(testFolder).createContent(CMISUtil.DocumentType.TEXT_PLAIN);
```
Comment (kavitshah-gl) [inline amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/rules/FileVersionAsRecordRuleTest.java:195]: Record version should be checked and then validated.

File: amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/rules/MoveToRuleOnFoldersTest.java [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+    private Record electronicRecord;
+    private UnfiledContainer unfiledContainer;
+
+    private UserModel rmAdmin;
+    public RecordCategory RecordCategoryOne;
+    public RestNodeModel RecordCategoryCopy;
+    private RecordCategoryChild recordFolder;
+    public static final String RECORD_FOLDER_ONE = "record-folder-one";
+    private final String TEST_PREFIX = generateTestPrefix(MoveToRuleOnFoldersTest.class);
+    // private final String RM_ADMIN = TEST_PREFIX + "rm_admin";
```
Comment (kavitshah-gl) [inline amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/rules/MoveToRuleOnFoldersTest.java:85]: Remove commented line.

File: amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/rules/MoveToRuleOnFoldersTest.java [hunk 9]
```diff
@@ -0,0 +1,219 @@
...
+import static org.alfresco.rest.rm.community.utils.CoreUtil.toContentModel;
+import static org.alfresco.rest.rm.community.utils.FilePlanComponentsUtil.*;
+import static org.alfresco.utility.data.RandomData.getRandomAlphanumeric;
+import static org.alfresco.utility.data.RandomData.getRandomName;
+import static org.alfresco.utility.report.log.Step.STEP;
+import static org.springframework.http.HttpStatus.*;
+import static org.testng.Assert.assertNotNull;
+
+
+public class MoveToRuleOnFoldersTest extends BaseRMRestTest{
```
Comment (kavitshah-gl) [inline amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/rules/MoveToRuleOnFoldersTest.java:69]: Refactor the entire test case.

**Review by kavitshah-gl — CHANGES_REQUESTED** [review]

**Review by kavitshah-gl — APPROVED** [review]
LGTM.

**Comment by Sbisht19** [comment]
merged with master branch hence closing the PR.

**Pull request merged**